src/tests/unit/tests/background/stores/assessment-card-selection-store.test.ts

```diff
@@ -85,6 +87,181 @@ describe('AssessmentCardSelectionStore Test', () => {
         expectedState = cloneDeep(defaultState);
     });
 
+    describe('initialize', () => {
+        it('sets the state based on param', () => {
+            const testObject = new AssessmentCardSelectionStore(
+                new AssessmentCardSelectionActions(),
+                null,
+                null,
+                null,
+                false,
+                '',
+            );
+            testObject.initialize(initialState);
+
+            expect(testObject.getState()).toEqual(expectedState);
+        });
+
+        it('sets the state based on persisted state if no param', () => {
+            const testObject = new AssessmentCardSelectionStore(
+                new AssessmentCardSelectionActions(),
+                initialState,
+                null,
+                null,
+                true,
+                '',
+            );
+            testObject.initialize();
+
+            expect(testObject.getState()).toEqual(expectedState);
+        });
+
+        it('sets the state to default state if no param or persisted state', () => {
+            const testObject = new AssessmentCardSelectionStore(
+                new AssessmentCardSelectionActions(),
+                null,
+                null,
+                null,
+                false,
+                '',
+            );
+            testObject.initialize();
+
+            expect(testObject.getState()).toEqual({});
+        });
+    });
+
+    describe('onAssessmentStoreChanged', () => {
+        it('sets the store state with assessment data', async () => {
+            const payload: AssessmentStoreChangedPayload = {
+                assessmentStoreData: createAssessmentStoreDataWithStatus(ManualTestStatus.FAIL),
+            };
+
+            const storeTester =
+                createStoreForAssessmentCardSelectionActions(
+                    'assessmentStoreChanged',
+                ).withActionParam(payload);
+            await storeTester.testListenerToBeCalledOnce(null, expectedState);
```

Comment:
This test/ test name seems misleading- the title implies that we're setting the state based on the assessment data but we end up verifying that the final state is the expectedState, which is just the default state. Am I misunderstanding?

Reply:
Ah I see, thanks for the clarification!